knotx-core/src/main/java/io/knotx/dataobjects/Fragment.java

```diff
@@ -36,44 +38,62 @@
   private static final String KNOTS_KEY = "_KNOTS";
   private static final String CONTENT_KEY = "_CONTENT";
   private static final String CONTEXT_KEY = "_CONTEXT";
+  private static final String FALLBACK_KEY = "_FALLBACK";
 
-  private final List<String> knots;
+
+  private final List<Knot> knots;
   private final JsonObject context;
   private String content;
+  private String fallback;
 
   public Fragment(JsonObject fragment) {
-    this.knots = fragment.getJsonArray(KNOTS_KEY).stream().map(String::valueOf)
-        .collect(Collectors.toList());
+    JsonArray knotsArray = fragment.getJsonArray(KNOTS_KEY);
+    this.knots = new ArrayList<>();
```

Comment:
How about:
```suggestion
    this.knots= knotsArray.stream()
        .map(entry -> new Knot((JsonObject) entry))
        .collect(Collectors.toList());
```

If  you don't like doing this streams way, at least please initialize list with size `new ArrayList<>(knotsArray.size());`

Reply:
I like it, changed.